### ISSUE TITLE
export default @decl class {} is parsed as a class expression

### ISSUE THREAD

**Issue by nicolo-ribaudo** [babel/babel#7123]
### Input Code

```js
export default @dec class A {}
```

https://astexplorer.net/#/gist/2688ce655edf069660a7286477acd1fe/289582be3068e25daa64d745f6fa71275ffca05c

### Expected Behavior
It should be a classs declaration

**Comment by nveenjain** [comment]
Hey @nicolo-ribaudo , I would love to work on this, I was going through parser file in babylon, it felt overwhelming, i don't know how to start there. Could you just point me from where should i start? or which file should i start? Also if there is similar guide for parser as there is for plugin? Thanks.

**Comment by nicolo-ribaudo** [comment]
You should look at https://github.com/babel/babel/blob/master/packages/babylon/src/parser/statement.js#L1357-L1376, which parses the code after `export default`.

**Comment by joaomcarvalho** [comment]
@nicolo-ribaudo is that a good first issue? I would like to help.

**Comment by nicolo-ribaudo** [comment]
@nveenjain Have you already started working on this issue?

**Comment by nveenjain** [comment]
Yes @nicolo-ribaudo , i have started working on it. Though i've having some doubts, i was trying to clear them out manually using debugger. I was think to put an if statement here 
which checks for decorator too.. https://github.com/babel/babel/blob/master/packages/babylon/src/parser/statement.js#L1371

**Comment by nicolo-ribaudo** [comment]
Yes, it's correct :+1

**Comment by nveenjain** [comment]
@nicolo-ribaudo , I've submitted the PR, thanks for the patience, babylon was new for me. Learnt a lot while solving the issue :smiley:

**Comment by jsg2021** [comment]
@hzoo Did this land yet? I'm still observing it in 8.2.2.
